### DISCUSSION TITLE
ERR when run noahmp.3.6 with DA GLASS Albedo data

### DISCUSSION THREAD

**Discussion by zhengzhenk** [NASA-LIS/LISF#1311] [Q&A]
Hi LIS Community,

I am trying to run the noahMP.3.6 model and assimilate GLASS Albedo data with the LISF-7.4. However, I encountered an error message "[ERR] Error: StateGet Observation02 Stopping" during the model run. Although I have identified the code location where the model stops, I am unable to resolve this issue. Could anyone please suggest a solution? I have attached my lis.config file, lislog file, and the code at the error termination point below.

Any suggestions would be greatly appreciated.

Thank you,
Zhenkang Zheng

[run_DA_GLASSAlbedo.config.txt](https://github.com/NASA-LIS/LISF/files/11234975/run_DA_GLASSAlbedo.config.txt)

[lislog_DA_Albedo041401.0000.txt](https://github.com/NASA-LIS/LISF/files/11234976/lislog_DA_Albedo041401.0000.txt)

[read_GLASSalbedo.F90.txt](https://github.com/NASA-LIS/LISF/files/11234977/read_GLASSalbedo.F90.txt)

**Comment by dmocko** [comment]
Can you please share these files?

```
Observation attributes file:                       ./attrs_Albedo_GLASS/obs_attribs.txt
Observation perturbation attributes file:          ./attrs_Albedo_GLASS/obs_pertattribs.txt
```

There are two observations required for the GLASS albedo - black sky (direct) and white sky (diffuse).

You may have only one of these listed, which is why you are seeing this error.

**Reply by zhengzhenk** [reply]
@dmocko 
Thank you very much for your response.  Previously, I had only listed one variable in both the "Observation attributes file" and the "Observation perturbation attributes file". Following your suggestion, I listed both variables "black sky" and "white sky" in these files, and changed the value of "observation types" to 2 in the configure file as follows:

`Data assimilation number of observation types:       2`

This solved the previous error.However, unfortunately, I encountered a new problem, and the error message is as follows:

```
[INFO] Assimilating Observations using EnKF for DA instance 1
[ERR] ESMF_AttributeGet for Grid Number in generateObservations Stopping.
[ERR] endrun is being called.
```

I tried to solve this problem but was not successful. I found the place where the model stopped running in LISF_public_release_7.4.0\lis\dataassim\algorithm\ensrf\ensrf_Mod.F90, but I could not solve the problem based on the error message. I would like to ask for help to figure out the reason for this new error.
Meanwhile, I noticed that every time I run LIS on my workstation, a warning message is printed on the command line:

```
At line 82859 of file /home/zhengzk/soft/esmf-ESMF_8_0_1/src/Superstructure/AttributeAPI/interface/ESMF_Attribute.F90
Fortran runtime warning: An array temporary was created
```

Although this warning did not affect my LIS running results before, I wonder if it is also the reason for the current error?
I have listed all my configure, attribs, and pertattribs files used in the assimilation process below, as well as the lislog file and the file where the model produced error messages.

Thank you,
Zhenkang Zheng

[run_DA_Albedo_GLASStest.config.txt](https://github.com/NASA-LIS/LISF/files/11413957/run_DA_Albedo_GLASStest.config.txt)
[forcing_attribs.txt](https://github.com/NASA-LIS/LISF/files/11413958/forcing_attribs.txt)
[forcing_pertattribs.txt](https://github.com/NASA-LIS/LISF/files/11413959/forcing_pertattribs.txt)
[obs_attribs.txt](https://github.com/NASA-LIS/LISF/files/11413972/obs_attribs.txt)
[obs_pertattribs.txt](https://github.com/NASA-LIS/LISF/files/11413973/obs_pertattribs.txt)
[state_attribs.txt](https://github.com/NASA-LIS/LISF/files/11413974/state_attribs.txt)
[state_pertattribs.txt](https://github.com/NASA-LIS/LISF/files/11413975/state_pertattribs.txt)
[lislog_test_Albedo_modis.0000.txt](https://github.com/NASA-LIS/LISF/files/11413965/lislog_test_Albedo_modis.0000.txt)
[ESMF_Attribute.F90.txt](https://github.com/NASA-LIS/LISF/files/11413966/ESMF_Attribute.F90.txt)
